qless-java/src/main/java/com/moz/qless/Job.java

```diff
@@ -114,31 +114,32 @@ public void complete(final String nextQueueName, final Map<String, Object> opts)
       throws IOException {
     if (null == nextQueueName) {
       this.client.call(
-          LuaCommand.COMPLETE.toString(),
+          LuaCommand.COMPLETE,
           this.jid,
           this.client.workerName(),
           this.queueName,
           JsonUtils.stringify(this.data));
     } else {
       this.client.call(
-          LuaCommand.COMPLETE.toString(),
+          LuaCommand.COMPLETE,
           this.jid,
           this.client.workerName(),
           this.queueName,
           JsonUtils.stringify(this.data),
-          LuaConfigParameter.NEXT.toString(),
+          LuaConfigParameter.NEXT,
           nextQueueName,
-          LuaConfigParameter.DELAY.toString(), MapUtils.get(opts,
+          LuaConfigParameter.DELAY, MapUtils.get(opts,
               LuaConfigParameter.DELAY.toString(),
-              ClientHelper.DEFAULT_DELAY),
-          LuaConfigParameter.DEPENDS.toString(),
+              ClientHelper.DEFAULT_DELAY.toString()),
+          LuaConfigParameter.DEPENDS,
           JsonUtils.stringify(MapUtils.getList(opts,
               LuaConfigParameter.DEPENDS.toString())));
     }
   }
 
-  public Object getDataField(final String key) {
-    return this.data.get(key);
+  @SuppressWarnings("unchecked")
+  public <T> T getDataField(final String key) {
```

Comment:
Also, there's some newer (java 7) generic type inference things that might be applicable here (I'm thinking the `final List<Thing> list = new ArrayList<>();` stuff.)

I wonder if that could fly here.  Maybe @b4hand knows?